lib/galaxy/managers/notification.py

```diff
@@ -0,0 +1,100 @@
+from typing import (
+    Any,
+    Dict,
+    List,
+)
+
+from sqlalchemy import false
+
+from galaxy import model
+from galaxy.exceptions import (
+    Conflict,
+    ObjectAttributeMissingException,
+    ObjectNotFound,
+)
+from galaxy.managers.base import decode_id
+from galaxy.managers.context import ProvidesAppContext
+from galaxy.schema.fields import EncodedDatabaseIdField
+from galaxy.structured_app import MinimalManagerApp
+from galaxy.web import url_for
+
+
+class NotificationManager:
+    """Interface/service object shared by controllers for interacting with notifications."""
+
+    def __init__(self, app: MinimalManagerApp) -> None:
+        self._app = app
+
+    def index(self, trans: ProvidesAppContext):
+        """
+        Displays a collection (list) of groups.
+        """
+        rval = []
+        for notification in trans.sa_session.query(model.Notification).filter(model.Notification.deleted == false()):
+            item = notification.to_dict(value_mapper={"id": trans.security.encode_id})
+            encoded_id = trans.security.encode_id(notification.id)
+            item["url"] = url_for("notification", id=encoded_id)
+            rval.append(item)
+        return rval
+
+    def create(self, trans: ProvidesAppContext, payload: Dict[str, Any]):
+        """
+        Creates a new notification.
+        """
+        message_text = payload.get("Message", None)
+        if message_text is None:
+            raise ObjectAttributeMissingException("Missing required message")
+
+        notification = model.Notification(message_text=message_text)
+        trans.sa_session.add(notification)
+        encoded_user_ids = payload.get("user_ids", [])
+        users = self._get_users_by_encoded_ids(trans, encoded_user_ids)
+        trans.app.security_agent.set_entity_notification_associations(users=[users], notification=notification)
+        trans.sa_session.flush()
+        item = group.to_dict(view="element", value_mapper={"id": trans.security.encode_id})
+        item["url"] = url_for("notification", id=encoded_id)
```

Comment:
We also want to eliminate usage of `url_for`, is there a specific use you have in mind for the url ?

Reply:
I was thinking that notifications will be posted to a specific /notification/ page for the user.

Reply:
Yes, but why do you need the URL ?